Product/Production/Services/DocumentDataSubmissionCore/src/test/java/gov/hhs/fha/nhinc/docdatasubmission/audit/transform/DocDataSubmissionAuditTransformsTest.java

```diff
@@ -0,0 +1,233 @@
+        Properties webContextProperties = new Properties();
+        webContextProperties.setProperty(NhincConstants.WEB_SERVICE_REQUEST_URL, WS_REQUEST_URL);
+        webContextProperties.setProperty(NhincConstants.REMOTE_HOST_ADDRESS, REMOTE_IP);
+        webContextProperties.setProperty(NhincConstants.INBOUND_REPLY_TO, NhincConstants.WSA_REPLY_TO);
+        DocDataSubmissionAuditTransforms transforms = new DocDataSubmissionAuditTransforms() {
+            @Override
+            protected String getLocalHostAddress() {
+                return LOCAL_IP;
+            }
+
+            @Override
+            protected String getRemoteHostAddress(Properties webContextProeprties) {
+                if (webContextProeprties != null && !webContextProeprties.isEmpty() && webContextProeprties.
+                    getProperty(NhincConstants.REMOTE_HOST_ADDRESS) != null) {
+                    return webContextProeprties.getProperty(NhincConstants.REMOTE_HOST_ADDRESS);
+                }
+                return AuditTransformsConstants.ACTIVE_PARTICIPANT_UNKNOWN_IP_ADDRESS;
+            }
+
+            @Override
+            protected String getWebServiceUrlFromRemoteObject(NhinTargetSystemType target, String serviceName) {
+                return WS_REQUEST_URL;
+            }
+        };
+
+        RegisterDocumentSetRequestType request = createRegisterDocumentSetRequestType();
+        RegistryResponseType response = new RegistryResponseType();
+        AssertionType assertion = createAssertion();
+        LogEventRequestType auditResponse = transforms.transformResponseToAuditMsg(request, response, assertion, null,
+            NhincConstants.AUDIT_LOG_INBOUND_DIRECTION, NhincConstants.AUDIT_LOG_NHIN_INTERFACE,
+            Boolean.FALSE, webContextProperties, NhincConstants.NHINC_XDR_SERVICE_NAME);
+        testGetEventIdentificationType(auditResponse, NhincConstants.NHINC_XDR_SERVICE_NAME, Boolean.FALSE);
+        testGetActiveParticipantDestination(auditResponse, Boolean.FALSE, webContextProperties, WS_REQUEST_URL);
+        testAuditSourceIdentification(auditResponse.getAuditMessage().getAuditSourceIdentification(), assertion);
+        testGetActiveParticipantSource(auditResponse, Boolean.FALSE, webContextProperties, REMOTE_IP);
+        assertParticipantObjectIdentification(auditResponse.getAuditMessage());
+        assertEquals("AuditMessage.Response direction mismatch", auditResponse.getDirection(),
+            NhincConstants.AUDIT_LOG_INBOUND_DIRECTION);
+        assertEquals("AuditMessage.Response ServiceName mismatch", auditResponse.getEventType(),
+            NhincConstants.NHINC_XDR_SERVICE_NAME);
+    }
+
+    @Override
+    protected AuditTransforms<RegisterDocumentSetRequestType, RegistryResponseType> getAuditTransforms() {
+        return new DocDataSubmissionAuditTransforms();
+    }
+
+    private void assertParticipantObjectIdentification(AuditMessageType auditMsg) {
```

Comment:
Optional: For sake of reading, can extract the multiple "auditMsg.getParticipantObjectIdentification().get(X)" into a local variable to shorten these a bit.